### PR TITLE
Setup scripts

### PR DESCRIPTION
**Pull request by gwsii**
Initial version of some setup scripts that should help workshop participants get software tools installed more quickly. Includes:

1) Script to check Node JS version and install dependencies in sub-projects
2) Script to install AWS-CLI
3) Script to install JQ (JSON Query)

Scripts will also take the necessary steps of installing homebrew, if the user does not already have it installed, to deliver JQ.

### CODE REVIEW

File: setup-nodejs.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+function open_node_download {
+    echo "Dohttps://nodejs.org/en/download/"
```
Comment (13scoobie) [inline setup-nodejs.sh]: is this a typo? - Dohttps?

Comment (gwsii) [inline setup-nodejs.sh]: Yes, thanks. It's a typo for some debug code; will clean it up.

Comment (gwsii) [inline setup-nodejs.sh]: Yes, thanks. It's a typo for some debug code; will clean it up.

**Comment by gwsii** [comment]
@13scoobie Thanks for the suggestion on the automated install of Node JS. Given that there is a choice between using NodeJS directly and NVM, I wanted the user to take that on manually, with only providing a link to where you can get NodeJS. We can provide scripts to install either, and that would be helpful, I agree.

**Comment by gwsii** [comment]
Going to merge these changes because re-ordering the steps in Lesson 0 should really help avoid confusion.

**Pull request merged**